src/amqp-client/connection.cr

```diff
@@ -133,13 +137,19 @@ class AMQP::Client
       end
     end
 
-    def close(msg = "")
+    def close(msg = "", no_wait = true)
       return if @closed
       Log.debug { "Closing connection" }
       write Frame::Connection::Close.new(200_u16, msg, 0_u16, 0_u16)
-      @closed = true
+      unless no_wait
+        frame = @reply_frames.receive
+        frame.as?(Frame::Connection::CloseOk) || raise UnexpectedFrame.new(frame)
```

Comment:
isn't `@reply_frames` a poor name if only `CloseOk` is expected on it?

Reply:
or if it can be used for other things, loop until we get the closeok and throw away anything else.